service/service.py

```diff
@@ -349,6 +349,29 @@ def deliver_item(order_id, item_id):
 
     return make_response(jsonify(order.serialize()), status.HTTP_200_OK)
 
+######################################################################
+# DELIVER AN ORDER
+######################################################################
+@app.route("/orders/<int:order_id>/deliver", methods=["PUT"])
+def deliver_orders(order_id):
+    """ deliver all the items of the Order that have not being delivered yet """
+    app.logger.info("Request to deliver order with id: %s", order_id)
+    order = Order.find(order_id)
+    if not order:
+        raise NotFound("Order with id '{}' was not found.".format(order_id))
+
+    shipped_delivered_canceled_orders = 0
+    for i in range(len(order.order_items)):
+        if order.order_items[i].status in ["PLACED", "CANCELLED"]:
```

Comment:
Even if ONE order item has a status of "PLACED" we need to throw a DataValidationError since an order item cannot be delivered unless it is shipped. 

In case it is cancelled, we cannot deliver it, and hence, only if all orders are "CANCELLED" then we need to throw a DataValidationError. 

These will be two separate conditions.

If all order items are either SHIPPED, DELIVERED or CANCELLED we return a 202, EXCEPT the case mentioned above i.e. all order items are cancelled in that case we return 400.